src/conv.rs

```diff
@@ -179,6 +179,17 @@ pub fn map_origin3d(native: &native::WGPUOrigin3D) -> wgt::Origin3d {
     }
 }
 
+pub fn map_adapter_options<'a>(
+    options: &native::WGPURequestAdapterOptions,
+    extras: Option<&native::WGPUAdapterExtras>,
+) -> (Option<id::SurfaceId>, u32) {
+    if let Some(extras) = extras {
+        (Some(options.compatibleSurface), extras.backendBits as u32)
```

Comment:
why is this always `Some`?

Reply:
Because Rust thinks `options.compatibleSurface` is `SurfaceId`, not `Option<SurfaceId>`. I suspect we need to apply similar trickery as earlier to tell bindgen?

It *does* work now. I.e. I set `options.comatibleSurface` to null (when calling the C-api). So maybe the `SurfaceId` value is somehow checked down the road? Maybe it is just ignored if it does not match an existing surface?

Reply:
Yes, definitely, we need a similar trickery every time there is an Option<Id>